pkg/gateway/gateway.go

```diff
@@ -9,22 +9,32 @@ import (
 	"strings"
 )
 
-type Request struct {
-	UriPath string
-}
-
 func route(req *Request, p proxy.Proxy, d datastore.DataStore) (io.ReadCloser, int) {
 	parts := strings.SplitAfter(strings.TrimLeft(req.UriPath, "/"), "/")
 	s, serr := d.GetService(strings.TrimRight(parts[0], "/"))
 	if serr != nil {
 		return ioutil.NopCloser(strings.NewReader(serr.Error())), http.StatusInternalServerError
 	}
-	body, err, statusCode := p.GetRequest("http://" + s.Host + ":" + s.Port + "/" + strings.Join(parts[1:], ""))
+	q := ""
+	if req.Query != nil {
```

Comment:
NIT: it's a little confusing to read these double loops for appending params. might be a good func breakout in the future.

Reply:
Agreed, ill break it out.